theme/js/components/search/search.vue

```diff
@@ -0,0 +1,293 @@
+          class="mt-md"
+        />
+      </ul>
+      <div v-else>
+        <Empty
+          wide
+          :queryString="queryString"
+          :cta="$t('@@Réinitialiser les filtres')"
+          :copy="
+            $t(
+              '@@Nous n’avons pas de jeu de données correspondant à votre requête'
+            )
+          "
+          :copyAfter="
+            $t(
+              '@@Vous pouvez essayer de réinitialiser les filtres pour agrandir votre champ de recherche.'
+            )
+          "
+          :onClick="() => resetFilters()"
+        />
+      </div>
+    </transition>
+  </section>
+</template>
+
+<script>
+import config from "../../config";
+import SearchInput from "./search-input";
+import Suggestor from "./suggestor";
+import Rangepicker from "./rangepicker";
+import Dataset from "../dataset/search-result";
+import Loader from "../dataset/loader";
+import Empty from "./empty";
+import Pagination from "../pagination/pagination";
+import { generateCancelToken } from "../../plugins/api";
+import filterIcon from "svg/filter.svg";
+
+import queryString from "query-string";
+
+export default {
+  components: {
+    "search-input": SearchInput,
+    Rangepicker,
+    Suggestor,
+    Dataset,
+    Empty,
+    Loader,
+    Pagination,
+  },
+  created() {
+    this.filterIcon = filterIcon;
+
+    //Update facets from URL on page load for deep linking
+    const url = new URL(window.location);
+    const searchParams = queryString.parse(url.search);
+
+    this.facets = searchParams;
+    this.search();
+  },
+  watch: {
+    facets: {
+      deep: true,
+      handler(val) {
+        //Update URL to match current facets value for deep linking
+        let url = new URL(window.location);
+        const searchParams = queryString.stringify(val, { skipNull: true });
+        url.search = searchParams;
+        history.pushState(null, "", url);
+      },
+    },
+  },
+  data() {
+    return {
+      extendedForm: false, //On desktop, extended form is simply another row of filters. On mobile, form is hidden until extendedForm is triggered
+      results: [],
+      loading: false,
+      currentRequest: null,
+      page_size: 20,
+      current_page: 1,
+      total_results: 0,
+      queryString: "",
+      facets: {},
+    };
+  },
+  computed: {
+    //Url for doing the same search (queryString only) on the reuse page
+    reuseUrl: function () {
+      return (
+        config.values.reuseUrl +
+        (this.queryString ? "?q=" + this.queryString : "")
+      );
+    },
+    //Is any filter active ?
+    isFiltered: function () {
+      return Object.keys(this.facets).some(
+        (key) => this.facets[key]?.length > 0
+      );
+    },
+  },
+  methods: {
+    handleSearchChange(input) {
+      this.queryString = input;
+      this.search();
+    },
+    //Called on every facet selector change, updates the `facets.xxx` object then searches with new values
+    handleSuggestorChange(facet) {
+      const that = this;
```

Comment:
avoid this (or that) by using arrow method?

Reply:
You can't touch `this` ! Turns out you can.
I'm so used to VueJS preventing you from ever using arrow functions that I stopped using them altogether. In React, it's the other way around